src/mpid/ch4/shm/ipc/src/ipc_init.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpidimpl.h"
+#include "ipc_noinline.h"
+#include "ipc_control.h"
+#include "shm_control.h"
+
+static void regist_shm_ctrl_cb(void)
+{
+    MPIDI_SHM_ctrl_reg_cb(MPIDI_SHM_IPC_SEND_CONTIG_LMT_RTS, &MPIDI_IPC_send_contig_lmt_rts_cb);
+    MPIDI_SHM_ctrl_reg_cb(MPIDI_SHM_IPC_SEND_CONTIG_LMT_FIN, &MPIDI_IPC_send_contig_lmt_fin_cb);
+}
+
+int MPIDI_IPC_mpi_init_hook(int rank, int size, int *tag_bits)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IPC_MPI_INIT_HOOK);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPC_MPI_INIT_HOOK);
+
+#ifdef MPL_USE_DBG_LOGGING
+    MPIDI_CH4_SHM_IPC_GENERAL = MPL_dbg_class_alloc("SHM_IPC", "shm_ipc");
+#endif
+
+    regist_shm_ctrl_cb();
+
+    MPIDI_IPC_global.node_group_ptr = NULL;
+
+    if (MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE != -1) {
+        mpi_errno = MPIDI_IPC_xpmem_mpi_init_hook(rank, size, tag_bits);
+        MPIR_ERR_CHECK(mpi_errno);
+    }
```

Comment:
```suggestion
    mpi_errno = MPIDI_IPC_xpmem_mpi_init_hook(rank, size, tag_bits);
    MPIR_ERR_CHECK(mpi_errno);
```

Reply:
We usually initialize all modules that are compiled in, even if they are not used based on some CVAR setting.

Reply:
This comment is not addressed yet. Same for finalize_hook